enhancements/monitoring/alert-overrides.md

```diff
@@ -0,0 +1,281 @@
+  - TBD
+creation-date: 2022-02-08
+last-updated: 2022-02-08
+tracking-link: https://issues.redhat.com/browse/MON-1985
+status: implementable
+---
+
+# User-Defined Alerting and Relabeling Rules
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The core OpenShift components ship a large number of Prometheus alerting rules.
+A 4.10-nightly cluster on AWS currently has 175 alerts defined by default, and
+enabling additional add-ons increases that number.  Still, users have repeatedly
+asked for a supported method of adding additional custom rules, adjusting
+thresholds, or adding and modifying labels on existing platform alerting rules
+-- for example, changing severity labels.
+
+These modifications are currently not possible, because the alerts are managed
+by operators and cannot be modified, and the addition of custom rules is
+explicitly unsupported at the moment.  This proposal outlines a solution for
+allowing additional user-defined rules and relabeling operations on existing
+platform alerting rules.
+
+## Motivation
+
+OpenShift is an opinionated platform, and the default alerting rules have of
+course been carefully crafted to match what we think the majority of users need.
+Nevertheless, users have repeatedly asked for the ability to modify alerting
+rules to adjust thresholds, or to simply add and modify labels to aide in
+routing alert notifications, as well as a supported method of adding additional
+alerting rules.
+
+Users currently have the ability to add additional rules from a technical
+standpoint -- by simply creating new `PrometheusRule` objects -- but this is
+[explicitly unsupported][1] at the moment.  The reason being that these
+resources are handled directly by the prometheus-operator, making it difficult
+to perform OpenShift specific defaulting and validation.  A concrete example
+being that there is no way to prevent users from creating new recording rules
+that may overwhelm the platform Prometheus instance.
+
+The goal of this proposal is to enable the addition of custom alerting rules,
+and limited ability to override labeling of existing platform alerting rules via
+resources under the direct control of the cluster-monitoring-operator.
+
+### Goals
+
+- Give users a supported and documented method to add additional rules, in a way
+  that allows the cluster-monitoring-operator to provide its own defaulting,
+  validation, telemetry, etc.
+- Allow users to apply relabeling operations to alerting rules shipped by the
+  platform.
+- Resources shipped by operators remain unmodified.  No action needed from
+  OpenShift developers.
+
+### Non-Goals
```

Comment:
Should we add that the modified alerts will not be reflected in the OCP console (e.g. it's purely for alert routing)?

Reply:
I added a note about this under the proposal section.